Sources/core/objects/Human.cpp

```diff
@@ -76,12 +78,33 @@ void Human::GenerateInterfaceForFrame()
 
 bool Human::TryMove(Dir direct)
 {
+    VDir pos;
+    if (pulled_object_)
+    {
+        if (!CanTouch(pulled_object_))
+        {
+            StopPull();
+        }
+        else
+        {
+            pos.x = GetX() - pulled_object_->GetX();
+            pos.y = GetY() - pulled_object_->GetY();
+        }
+    }
     if (IMob::TryMove(direct))
     {   
         if (owner->GetItem<Shard>().IsValid())
         {
             PlaySoundIfVisible("glass_step.ogg", GetOwner().Id());
         }
+        if (pulled_object_ && std::abs(pos.x) + std::abs(pos.y) < 2)
```

Comment:
It is better to place brackets around `std::abs(pos.x) + std::abs(pos.y) < 2`:
```
if (pulled_object_ && (std::abs(pos.x) + std::abs(pos.y) < 2))
```
because it is better to directly express operations priority in complicated expressions.